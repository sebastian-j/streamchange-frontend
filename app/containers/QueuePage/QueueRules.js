import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import {
  makeSelectCapacity,
  makeSelectQueueCommand,
  makeSelectTTI,
  makeSelectTTK,
  makeSelectWidgetCode,
} from './selectors';
import {
  changeCapacity,
  changeQueueCommand,
  changeTTI,
  changeTTK,
  changeWidgetCode,
} from './actions';
import Panel from '../../components/Panel';
import PanelTitle from '../../components/Panel/PanelTitle';
import StyledTextField from '../../components/StyledTextField';
import AdFrame from '../../components/AdFrame';
import QueueWidgetDialog from './QueueWidgetDialog';

export const QueueRules = props => {
  const handleInputValueChange = event => {
    const { target } = event;
    const { value } = target;
    const { name } = target;

    if (value.length < 140 && name === 'command')
      props.changeQueueCommand(value);
    if (name === 'capacity') props.changeCapacity(value);
    if (name === 'timeToIdle') props.changeTTI(value);
    if (name === 'timeToKick') props.changeTTK(value);
    if (name === 'widgetCode') {
      props.changeWidgetCode(value);
      return;
    }
    localStorage.setItem(`queue-${name}`, value);
  };

  return (
    <Panel>
      <PanelTitle>
        <FormattedMessage {...messages.rulesPanelTitle} />
      </PanelTitle>
      <FormattedMessage {...messages.commandTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="command"
            onChange={handleInputValueChange}
            label={label}
            type="text"
            value={props.command}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.capacityTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="capacity"
            onChange={handleInputValueChange}
            label={label}
            type="number"
            value={props.capacity}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.timeToIdleTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="timeToIdle"
            onChange={handleInputValueChange}
            label={label}
            type="number"
            value={Number(props.timeToIdle)}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.timeToKickTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="timeToKick"
            onChange={handleInputValueChange}
            label={label}
            type="number"
            value={Number(props.timeToKick)}
            fullWidth
          />
        )}
      </FormattedMessage>
      <FormattedMessage {...messages.widgetCodeTextField}>
        {label => (
          <StyledTextField
            autoFocus
            margin="dense"
            name="widgetCode"
            onChange={handleInputValueChange}
            label={label}
            type="password"
            value={props.widgetCode}
            fullWidth
          />
        )}
      </FormattedMessage>
      <QueueWidgetDialog />
      <AdFrame />
    </Panel>
  );
};

QueueRules.propTypes = {
  changeCapacity: PropTypes.func.isRequired,
  changeQueueCommand: PropTypes.func.isRequired,
  changeTTI: PropTypes.func.isRequired,
  changeTTK: PropTypes.func.isRequired,
  changeWidgetCode: PropTypes.func.isRequired,
  capacity: PropTypes.number.isRequired,
  command: PropTypes.string,
  timeToIdle: PropTypes.number,
  timeToKick: PropTypes.number.isRequired,
  widgetCode: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  capacity: makeSelectCapacity(),
  command: makeSelectQueueCommand(),
  timeToIdle: makeSelectTTI(),
  timeToKick: makeSelectTTK(),
  widgetCode: makeSelectWidgetCode(),
});

export function mapDispatchToProps(dispatch) {
  return {
    changeCapacity: cap => dispatch(changeCapacity(cap)),
    changeQueueCommand: command => dispatch(changeQueueCommand(command)),
    changeTTI: s => dispatch(changeTTI(s)),
    changeTTK: s => dispatch(changeTTK(s)),
    changeWidgetCode: code => dispatch(changeWidgetCode(code)),
    dispatch,
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(QueueRules);
