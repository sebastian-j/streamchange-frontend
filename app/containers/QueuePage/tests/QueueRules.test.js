import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import QueueRules from '../index';
import configureStore from '../../../configureStore';

describe('<QueueRules />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <QueueRules />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
