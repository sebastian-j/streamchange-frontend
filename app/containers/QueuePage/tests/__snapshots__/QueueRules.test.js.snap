// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueueRules /> should render and match the snapshot 1`] = `
<div
  class="sc-bdvvtL bfyUtR"
>
  <h2
    class="sc-gsDKAQ chMgaO"
  >
    <span>
      Queue rules
    </span>
  </h2>
  <div
    class="MuiFormControl-root MuiTextField-root sc-dkPtRN jhVsva MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
      data-shrink="false"
    >
      Command for joining the queue
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        name="command"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiTextField-root sc-dkPtRN jhVsva MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
      data-shrink="true"
    >
      Queue capacity
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        name="capacity"
        type="number"
        value="10"
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiTextField-root sc-dkPtRN jhVsva MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
      data-shrink="true"
    >
      Time without any message to mark as inactive (sec)
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        name="timeToIdle"
        type="number"
        value="600"
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiTextField-root sc-dkPtRN jhVsva MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
      data-shrink="true"
    >
      Time without any message to kick (sec)
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        name="timeToKick"
        type="number"
        value="900"
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiTextField-root sc-dkPtRN jhVsva MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense Mui-focused Mui-focused"
      data-shrink="true"
    >
      Code for OBS widget with queue
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        name="widgetCode"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    style="display: inline-block;"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text sc-hBUSln kPtqzs"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span>
          Queue widget for OBS
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div>
    <div
      class="sc-bqiRlB dxwfnn"
    >
      <span>
        <span>
          This is a place for your advertisement. Prices starts from 10 PLN per day. Your ad will reach thousands of viewers.
        </span>
      </span>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span>
            Show the details
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
